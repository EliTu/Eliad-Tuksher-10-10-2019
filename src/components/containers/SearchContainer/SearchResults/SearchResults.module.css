@import '../../../../variables.css';

.SearchResultsStyles {
    z-index: 2;
    display: flex;
    width: inherit;
    height: 100px;
    border: var(--border-thick);
    border-color: var(--color-main-black-opaque);
}