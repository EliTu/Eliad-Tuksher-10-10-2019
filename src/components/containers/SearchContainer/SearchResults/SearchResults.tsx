import React, { PropsWithChildren, ReactText } from 'react';
import styles from './SearchResults.module.css';

interface IProps {
	resultList: {}[];
	searchValue: string | ReactText;
	isDisplayed: boolean;
}

const SearchResults: React.FC<IProps> = ({
	resultList,
	searchValue,
	isDisplayed,
}) => {
	const { SearchResultsStyles } = styles;
	return <div className={SearchResultsStyles}></div>;
};

export default SearchResults;
