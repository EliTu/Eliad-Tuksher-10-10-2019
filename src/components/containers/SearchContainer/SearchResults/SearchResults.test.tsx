import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import SearchResults from './SearchResults';
import initialShallowRender from '../../../../utilities/test-utilities/initialShallowRender';

describe('SearchResults component', () => {
	let component: ShallowWrapper;
	beforeEach(
		() =>
			(component = shallow(
				<SearchResults
					resultList={[
						{
							LocalizedName: 'London',
							Key: '1',
							Country: { LocalizedName: 'UK' },
							AdministrativeArea: { ID: 'LD' },
						},
						{
							LocalizedName: 'London',
							Key: '1',
							Country: { LocalizedName: 'UK' },
							AdministrativeArea: { ID: 'LD' },
						},
						{
							LocalizedName: 'London',
							Key: '1',
							Country: { LocalizedName: 'UK' },
							AdministrativeArea: { ID: 'LD' },
						},
						{
							LocalizedName: 'London',
							Key: '1',
							Country: { LocalizedName: 'UK' },
							AdministrativeArea: { ID: 'LD' },
						},
					]}
					searchValue={'Lon'}
					isDisplayed={false}
				/>
			))
	);

	it('should render without errors', () => {
		component.setProps({ isDisplayed: true });
		initialShallowRender(component, '.SearchResultsStyles');
	});

	it('should not be displayed if searchValue length is < 2, isDisplayed is false or there are no elements in the resultList', () => {
		component.setProps({ resultList: [] });
		expect(component).toMatchSnapshot();
		expect(component).not.toContainMatchingElement('div');

		component.setProps({
			resultList: [
				{
					LocalizedName: 'London',
					Key: '1',
					Country: { LocalizedName: 'UK' },
					AdministrativeArea: { ID: 'LD' },
				},
			],
			searchValue: 'a',
		});
		expect(component).not.toContainMatchingElement('div');

		component.setProps({ searchValue: 'Lond', isDisplayed: false });
		expect(component).not.toContainMatchingElement('div');
	});

	it('should be visible if searchValue length its > 2, isDisplayed is true and if there are elements in the resultList', () => {
		component.setProps({ isDisplayed: true });
		expect(component).toContainMatchingElement('div');
	});

	it('should render and display a list of results coming from the resultList prop', () => {
		component.setProps({ isDisplayed: true });
		expect(component.children()).toBeTruthy();
		expect(component.children().children().length).toBe(4);
	});
});
