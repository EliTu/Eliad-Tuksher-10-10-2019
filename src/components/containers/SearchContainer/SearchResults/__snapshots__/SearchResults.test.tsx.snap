// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component should not be displayed if searchValue length is < 2, isDisplayed is false or there are no elements in the resultList 1`] = `""`;

exports[`SearchResults component should render without errors 1`] = `
<div
  className="SearchResultsStyles"
>
  <ul>
    <li
      key="1"
    >
      London, LD, UK
    </li>
    <li
      key="1"
    >
      London, LD, UK
    </li>
    <li
      key="1"
    >
      London, LD, UK
    </li>
    <li
      key="1"
    >
      London, LD, UK
    </li>
  </ul>
</div>
`;
